### PR TITLE
Update frontend and site dependencies

### PR DESCRIPTION
**Pull request by trustin**
Modifications:

- Updated most frontend and site dependencies to their latest versions,
  except those that need non-trivial migration:
  - MUI
  - Gatsby (Updated from v2 to v3, but not to v4)
- Fixed all newly reported ESLint violations, except those that don't
  make much sense.
  - Updated the `eslintrc.js` accordingly.
- Merged `steps.module.less` into `antd-overrides.less` because it
  doesn't contain any component-specific styles.
- Worked around the problem where the CSS class names of Table of
  Contents contain a plus (+) sign, preventing `tocbot` from working
  correctly.
- Updated `gatsby-remark-draw-patched.js` so it embeds the SVG images
  rather than writing to external files, because Gatsby doesn't handle
  them correcly anymore.
- Forked the SVGBob rendering code from `skyrta` because `svgbob` command
  has been renamed to `svgbob_cli` some time ago.
- Disabled `license-webpack-plugin` on Windows due to a known issue.

Result:

- 0 vulnerability in `docs-client`
- Less vulnerabilities in `site`
- A contributor can build `site` from macOS on aarch64 (M1)

### CODE REVIEW

**Comment by minwoox** [comment]
Oops, could you check the build failure in Windows OS?
https://github.com/line/armeria/runs/5356288534?check_suite_focus=true#step:6:344

File: docs-client/package.json [hunk 1]
```diff
@@ -12,65 +12,64 @@
...
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-react": "^7.29.2",
+    "eslint-plugin-react-hooks": "^4.3.0",
+    "favicons-webpack-plugin": "^5.0.2",
     "file-loader": "^6.2.0",
-    "html-webpack-plugin": "^4.5.2",
-    "license-webpack-plugin": "^2.3.18",
-    "prettier": "^2.2.1",
+    "html-webpack-plugin": "^5.5.0",
+    "license-webpack-plugin": "^4.0.2",
```
[the remark this answers is not in the document]
Comment (trustin) [reply docs-client/package.json:63]: I have no idea, either. What are we supposed to do, then?

[the remark this answers is not in the document]
Comment (ikhoon) [reply docs-client/package.json:63]: I found that it is a breaking change in `license-webpack-plugin`.
It is a new feature of `webpack-dev-server`.
https://github.com/webpack/webpack-dev-server/issues/3689#issuecomment-921732308
We can remove the overlay with:
```js
devServer: {
    client: {
        overlay: false 
        // OR 
        overlay: {
            warnings: false,
            errors: true
        }
    }
}
```

**Review by jrhee17 — APPROVED** [review]
Thanks @trustin ! Looks good to me once @ikhoon 's comment is addressed 🙇 👍 🙇

File: docs-client/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (trustin) [reply docs-client/package.json:63]: Nice find! Updated.

**Comment by trustin** [comment]
> Oops, could you check the build failure in Windows OS?
> [line/armeria/runs/5356288534?check_suite_focus=true#step:6:344](https://github.com/line/armeria/runs/5356288534?check_suite_focus=true#step:6:344)

I set up the dev environment on Windows and it seems like there's a bug somewhere between WebPack and license-webpack-plugin. I disabled the plugin when running on Windows for now given that we never release on Windows.

**Pull request merged**